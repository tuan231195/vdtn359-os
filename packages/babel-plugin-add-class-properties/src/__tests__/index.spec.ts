import pluginTester from 'babel-plugin-tester';
import plugin from '..';

describe('plugin tests', () => {
	pluginTester({
		plugin,
		snapshot: true,
		babelOptions: {
			filename: __filename,
			presets: ['@babel/typescript'],
		},
		tests: [
			{
				title: 'does nothing when there is no class',
				code: `
			        import foo from './a'
			        const a = 1;
			        export default 'something else';
		        `,
			},
			{
				title: 'add properties information',
				code: `
			        class A {
			            a: number;
			            b: number;
			            c: number;
			            static d: string;
			        }
			    `,
			},
		],
	});
});
