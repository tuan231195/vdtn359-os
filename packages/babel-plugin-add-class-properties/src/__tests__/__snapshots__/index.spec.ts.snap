// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`plugin tests unknown plugin add properties information: add properties information 1`] = `

class A {
    a: number;
    b: number;
    c: number;
    static d: string;
}

      ↓ ↓ ↓ ↓ ↓ ↓

class A {
    static __PROPERTIES__ = ['a', 'b', 'c'];
    static __STATIC_PROPERTIES__ = ['d'];
}


`;

exports[`plugin tests unknown plugin does nothing when there is no class: does nothing when there is no class 1`] = `

import foo from './a'
const a = 1;
export default 'something else';

      ↓ ↓ ↓ ↓ ↓ ↓

const a = 1;
export default 'something else';


`;
